src/brewvoid.h

```diff
@@ -426,6 +435,44 @@ void brew() {
 
                 break;
         }
+
+        if (brewcounter > kBrewIdle && brewcounter < kWaitBrewOff) {
+            // Calculate time and weight difference
+            unsigned long timeDelta = currentMillistemp - prevFlowRateTime;
+            float weightDelta = weightBrew - prevFlowRateWeight;
+            // avoid flowrate flicker when scale flickers negatively
+            if (weightDelta < 0) {
+                weightDelta = 0;
+            }
+
+            // Calculate flow rate
+            // we only get a new weight measurement every intervalWeight ms, so we do not need to compute this more often!
+            if (timeDelta > intervalWeight) {
+                float currentFlowRate = weightDelta / (timeDelta / 1000.0); // Flow rate in g/s
+
+                // Update buffer
+                flowRateBuffer[flowRateBufferIndex] = currentFlowRate;
+                flowRateBufferIndex = (flowRateBufferIndex + 1) % flowRateBufferSize;
+                if (flowRateMeasurements < flowRateBufferSize) {
+                    flowRateMeasurements++;
+                }
+
+                // Calculate the average flow rate
+                float averageFlowRate = 0.0;
+                for (int i = 0; i < flowRateMeasurements; i++) {
+                    averageFlowRate += flowRateBuffer[i];
+                }
+                flowRate = averageFlowRate / flowRateBufferSize;
```

Comment:
This is basically a box car filter which is not a very good low pass filter. Maybe consider using an exponential moving average here as it has a much simpler implementation and faster response to recent changes. We already have an implementation in filterPressureValue() albeit with fixed parameters. You could make that method generic and use it here (e.g. with 0.01 and 0.99 as params).

Reply:
I changed it to use EMA -- will determine potentially more fitting params with the next shots tomorrow ;) The calculation is so simple that I found it easier to do it in place rather than making the existing method generic...